lib/bambora/json_client.rb

```diff
@@ -2,22 +2,61 @@
 
 module Bambora
   class JSONClient < Bambora::Client
-    def request(method:, path: '', params: {}, body: {}, headers: {})
-      resp = connection.request(
+    # Make a JSON Request.
+    #
+    # @example
+    #
+    #   client = Bambora::JSONClient(base_url: '...', api_key: '...', merchant_id: '...')
+    #
+    #   data = {
+    #    language: 'en',
+    #     card: {
+    #       name: 'Hup Podling',
+    #       number: '4030000010001234',
+    #       expiry_month: '12',
+    #       expiry_year: '23',
+    #       cvd: '123',
+    #     },
+    #   }
+    #
+    #   client.request(
+    #     method: :post,
+    #     path: 'v1/profiles',
+    #     body: data,
+    #     api_key: '...'
+    #   )
+    #   # => {
+    #   #      :code => 1,
+    #   #      :message => "Operation Successful",
+    #   #      :customer_code => "02355E2e58Bf488EAB4EaFAD7083dB6A",
+    #   #    }
+    #
+    # @param method [Symbol] Indicating request verb.
+    # @param path [String] Indicating request path.
+    # @param body [Hash] Data to be sent in the body of the request. This should be a Hash or an object that responds
+    #   to +to_hash+.
+    # @param api_key [String] Indicating the API Key to be used with the request.
+    #
+    # @return [Hash] Indicating success or failure of the operation.
+    def request(method:, path:, body: {}, api_key:)
+      request_options = {
         method: method,
         path: path,
-        params: params,
-        body: body.to_json.to_s,
-        headers: { 'Content-Type' => 'application/json' }.merge(headers),
-      )
-
+        body: request_body(body),
```

Comment:
I'm not sure why this stated happening, but Excon started to complain that the request body was not a string. I've added this method fro stringifying it.